src/System.Diagnostics.Metrics.Temp/Instrument.cs

```diff
@@ -29,6 +29,7 @@ namespace System.Diagnostics.Metrics
     public abstract class Instrument
     {
         internal ConcurrentDictionary<MeterListener, bool> Listeners = new ConcurrentDictionary<MeterListener, bool>();
+        internal KeyValuePair<MeterListener, bool>[] CachedListeners = new KeyValuePair<MeterListener, bool>[0];
```

Comment:
Use `Array.Empty<KeyValuePair<MeterListener, bool>>()` instead to [avoid memory allocation](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1825).

Reply:
The System.Diagnostics.Metrics.Temp will be replaced with the official .NET API soon.

Array.Empty<> is not available on all framework, and thus, the new KeyValuePair<>[0].  Plus, it is known this will cause an allocation, but it's not in the Hot Path, so much less of a concern.